sig-and-wg/sig-core/proposals/installer.md

```diff
@@ -0,0 +1,42 @@
+# Kyma Installer - dynamic installation 
+
+Created on 2018-08-23 by Krysian Cieślik (@crabtree) and Jakub Kabza (@jakkab).
+This document is a part of the [Kyma modularization proposal](./modularization.md).
+
+## Status
+
+Proposed on 2018-08-23.
+
+## Motivation
+
+Kyma installation is a complex, time-consuming process, involving multiple components. The procedure is followed every day by a large number of developers who test their features, and thus should easily handle different scenarios without modifications. This is where the dynamic installation comes in handy, unifying, simplifying and automating the process.
+
+## Goal
+
+This proposal aims to enhance developer experience by:
+
+- extending installation customizability 
+- reducing the number of different installation methods
+- avoiding redundant modifications to the Installer component
+
+## Current solution
+
+Kyma installation:
+
+- consists of hard-coded steps
+- uses a variety of installation methods throughout the process, such as custom shell scripts, Helm, and kubectl commands
+- stores overrides in a fixed struct that requires code modifications every time new parameter is introduced
+
+All things considered, the installation process lacks coherence and leaves little room for customization, requiring constant maintenance at the same time.
+
+## Suggested solution
+
+- Installer is a component that handles installation, update, and deletion of Helm charts.
+- Installer no longer supports Bash scripts.
+- In order to increase security, Docker container used to run Installer does not feature Bash at all.
+- Installer is provided with a descriptive and comprehensible list of Kyma components, specified in the `kyma-installation` custom resource.
+- The list is declarative. It is for the user to decide which components will be installed.
+- Each component on the list is identified by its name and Namespace.
+- Installer handles dependencies between components to a limited extent. It is up to the user to specify a relevant order of components on the list.
```

Comment:
could you elaborate a bit more about this point? what is the limited extend? maybe you have already an example yaml structure in mind that could visualize? what dependencies you actually have in mind? like that helm broker is dependent on service catalog and that I can somehow define that it is installed after service catalog installation?

Reply:
see discussion here: https://github.com/kyma-project/community/pull/26#discussion_r213219837

Reply:
so to make sure, by limited extend you mean that configuration of installed components will look similar to:
```
- components
      - service-catalog
      - helm-broker:
          requires:
            - service-catalog
      - console
```
Do we really expect that `It is up to the user to specify a relevant order of components on the list.` or is it a statement related to additional components that user owns?

Reply:
@derberg looking forward to your suggestions!

Reply:
so to make it clear, I don't like that dependencies are handled by a flat ordered list, I like approach as described in my example, so if console is not dependent on service-catalog, why it cannot be installed asynchronously. 

so change `Installer handles dependencies between components to a limited extent` to `Installer handles dependencies between components basing it on an order in a flat list. `
You can also check if it makes sense with @klaudiagrz

Reply:
`Installer handles dependencies between components basing on an order in a flat list.` 😉